src/lib/base.jl

```diff
@@ -101,6 +102,22 @@ end
   Base.sync_end(refs), _ -> foreach(t -> runadjoint(__context__, t), refs)
 end
 
+@adjoint! function Base.sync_end(ch::Channel)
+  c = grad_mut(__context__, ch)
+  b = grad_mut(c)
+  while !isempty(ch)
+    i = take!(ch)
+    put!(c, i)
+    put!(b, i)
+  end
+  Base.sync_end(c), _ -> begin
+    while !isempty(b)
+      t = take!(b)
+      runadjoint(__context__, t)
+    end
+  end
+end
```

Comment:
Can we have meaningful variable names?
I have no ide what any of these variables mean, and thus this is hard to reivew.

Reply:
Oh sure, yes, that should have been caught sooner.